.azure/pipelines/jobs/default-build.yml

```diff
@@ -41,217 +41,223 @@
   # jobDisplayName: '' - use agentOs by default.
-  artifacts:  []
-  buildDirectory: ''
-  buildScript: ''
+  artifacts: []
+  buildDirectory: ""
+  buildScript: ""
   installTar: true
   installNodeJs: true
   installJdk: true
   timeoutInMinutes: 180
+  testRunTitle: $(AgentOsName)-$(BuildConfiguration)-xunit
 
   # We need longer than the default amount of 5 minutes to upload our logs/artifacts. (We currently take around 5 mins in the best case).
   # This makes sure we have time to upload everything in the case of a build timeout - really important for investigating a build
   # timeout due to test hangs.
   cancelTimeoutInMinutes: 15
 
 jobs:
-- template: /eng/common/templates/job/job.yml
-  parameters:
-    name: ${{ coalesce(parameters.jobName, parameters.agentOs) }}
-    displayName: ${{ coalesce(parameters.jobDisplayName, parameters.agentOs) }}
-    dependsOn: ${{ parameters.dependsOn }}
-    ${{ if ne(parameters.condition, '') }}:
-      condition: ${{ parameters.condition }}
-    timeoutInMinutes: ${{ parameters.timeoutInMinutes }}
-    cancelTimeoutInMinutes: ${{ parameters.cancelTimeoutInMinutes }}
-    ${{ if and(eq(variables['System.TeamProject'], 'internal'), eq(parameters.agentOs, 'Windows'), eq(parameters.codeSign, 'true')) }}:
-      enableMicrobuild: true
-      enablePublishBuildAssets: true
-      enablePublishUsingPipelines: ${{ variables._PublishUsingPipelines }}
-    enablePublishTestResults: true # publish test results to AzDO (populates AzDO Tests tab)
-    enableTelemetry: true
-    helixRepo: dotnet/aspnetcore
-    helixType: build.product/
-    workspace:
-      clean: all
-    # Map friendly OS names to the right queue
-    # See https://github.com/dotnet/arcade/blob/master/Documentation/ChoosingAMachinePool.md
-    pool:
-      ${{ if eq(parameters.agentOs, 'macOS') }}:
-        vmImage: macOS-10.13
-      ${{ if eq(parameters.agentOs, 'Linux') }}:
-        vmImage: ubuntu-16.04
-      ${{ if eq(parameters.agentOs, 'Windows') }}:
-        ${{ if eq(variables['System.TeamProject'], 'public') }}:
-          name: NetCorePublic-Pool
-          ${{ if ne(parameters.isTestingJob, true) }}:
-            # Visual Studio Build Tools
-            queue: BuildPool.Server.Amd64.VS2019.BT.Open
-          ${{ if eq(parameters.isTestingJob, true) }}:
+  - template: /eng/common/templates/job/job.yml
+    parameters:
+      name: ${{ coalesce(parameters.jobName, parameters.agentOs) }}
+      displayName: ${{ coalesce(parameters.jobDisplayName, parameters.agentOs) }}
+      dependsOn: ${{ parameters.dependsOn }}
+      ${{ if ne(parameters.condition, '') }}:
+        condition: ${{ parameters.condition }}
+      timeoutInMinutes: ${{ parameters.timeoutInMinutes }}
+      cancelTimeoutInMinutes: ${{ parameters.cancelTimeoutInMinutes }}
+      ? ${{ if and(eq(variables['System.TeamProject'], 'internal'), eq(parameters.agentOs, 'Windows'), eq(parameters.codeSign, 'true')) }}
+      : enableMicrobuild: true
```

Comment:
Where is this syntax defined? It's not mentioned in https://docs.microsoft.com/en-us/azure/devops/pipelines/process/templates?view=azure-devops and the YAML 1.2 key / value syntax seems odd here.

Reply:
Wasn't changed in this PR, so 🤷‍♀

Reply:
This was changed in this PR. See https://github.com/dotnet/aspnetcore/pull/19575/files#diff-f3d39efe7528a28e4995c18938c8a3d3L78

Seems like something is doing lots of extra changes to the YAML files you're touching, making it hard to see what's really changing.